Tests/Functional/Domain/Repository/Product/TeaRepositoryTest.php

```diff
@@ -122,11 +122,11 @@ public function addAndPersistAllCreatesNewRecord(): void
         $this->subject->add($model);
         $this->persistenceManager->persistAll();
 
-        $databaseRow = $this->getDatabaseConnection()->selectSingleRow(
-            '*',
-            'tx_tea_domain_model_product_tea',
-            'uid = ' . $model->getUid()
-        );
+        $connection = $this->getConnectionPool()
+            ->getConnectionForTable('tx_tea_domain_model_product_tea');
+        $databaseRow = $connection->select(['*'], 'tx_tea_domain_model_product_tea', ['uid' => $model->getUid()])
+            ->fetchAssociative();
```

Comment:
To be compatible with TYPO3 10LTS with lowest dependencies, we need to use the deprecated method here (maybe with a comment).

Reply:
These are the requirements in the `composer.json` of `typo3/cms-core`:

- V10: `"doctrine/dbal": "~2.10.0 || ~2.11.2 || ~2.13.1",`
- V11: `"doctrine/dbal": "^2.13.5",`
- main: `"doctrine/dbal": "^3.2",`